### DISCUSSION TITLE
Test resources folder precendence

### DISCUSSION THREAD

**Discussion by romain-gilles-ultra** [com-lihaoyi/mill#3006] [Q&A]
Hi Mill,
It sounds like the `resources` has no precedence on the library it depends on...
Here an example
```
+- my-module
   +- src
      +- main
         +- java
         +- scala
         +- resources
            +- my-config.properties
      +- test
         +- java
         +- scala
         +- resources
            +- my-config.properties
```
When I run my test (JUnit5 => java code integration)  and I load the property file then I get the content from
```
src/main/resources/my-config.properties
```
But if I run the test in Intellij I get the correct behavior.
I wonder what is the precedence of the `resources` folder in the classpath?
Or maybe I'm doing something wrong :'(

Thanks

**Comment by lefou** [comment]
Depending on ordering of classpath entries is a call for trouble. If you assume a single file (e.g. you don't explicitly handle _all_ files/resources with a specific name) you should not have two of them on the classpath. The simplest solution is to use a different file name in the tests.

**Comment by lihaoyi** [comment]
> Depending on ordering of classpath entries is a call for trouble

Just to call this out again, if you deploy your application in the common "folder full of jars" format, exactly which jar appears first in the classpath would depend on filesystem ordering, which is variable across different operating systems and even different filesystems within the same operating system. That means which `application.conf` gets picked could very well differ unpredictably between your development environment, testing environment, and deployment environment.

In my work codebase we have a linter that fails if there is classpath conflicts, after getting bitten by this sort of problem one too many times. Perhaps we should add such a feature opt-in to Mill as well
